src/transport_performance/gtfs/validation.py

```diff
@@ -1088,6 +1107,16 @@ def _extended_validation(
             except NameError:
                 pass
 
+            # add a more detailed route_type decription
+            if "route_type" in impacted_rows.columns:
+                ROUTE_LKP = get_saved_route_type_lookup()
```

Comment:
`GtfsInstance.html_report` calls `_extended_validation()` which uses `get_saved_route_type_lookup()` .

`GtfsInstance.html_report` also calls `_plot_summary()` which separately calls `get_saved_route_type_lookup()` .

It would be better to create a class attribute for `GtfsInstance` on init, and to use this single table in both. This class attribute would be passed to every instance of `GtfsInstance` .

Eg:
```
class GtfsInstance:
    """Create a feed instance for validation, cleaning & visualisation."""
    ROUTE_LKP = get_saved_route_type_lookup()

    def __init__(
        self, gtfs_pth=here("tests/data/newport-20230613_gtfs.zip"), units="m"
    ):
		# Somewhere in init...
        ROUTE_LKP = GtfsInstance.ROUTE_LKP

```


Then to access it…

```
>>> foo = GtfsInstance()
>>> foo.ROUTE_LKP
   route_type                                               desc
0           0  Tram, Streetcar, Light rail. Any light rail or...
1           1  Subway, Metro. Any underground rail system wit...
2           2  Rail. Used for intercity or long-distance travel.
3           3  Bus. Used for short- and long-distance bus rou...
4           4  Ferry. Used for short- and long-distance boat ...
..        ...                                                ...
86       1505                              Licensed Taxi Service
87       1506                       Private Hire Service Vehicle
88       1507                                  All Taxi Services
89       1700                              Miscellaneous Service
90       1702                               Horse-drawn Carriage

[91 rows x 2 columns]
```

Reply:
@r-leyshon This should be resolved now with commit 289d4d2